### DISCUSSION TITLE
`Object.dispose` should fire `removed`?

### DISCUSSION THREAD

**Discussion by ShaMan123** [fabricjs/fabric.js#7447] [Ideas]
I think `Object.dispose` should fire `removed`.
It's easy to fix.
@asturur what do you think?

**Comment by ShaMan123** [comment]
#7448

**Comment by asturur** [comment]
I would think is the contrary.
Dispose is for freeing resources mainly.
You can dispose an object that is on the canvas ( and you will create issues )
You can remove an object from the canvas but not wanting to dispose it ( because you want to add it back later or move to another canvas ).
I do not think there should be correlation on those two functionalities

**Reply by ShaMan123** [reply]
I understand. Agreed
